bats/tests/profile/create-profile-output.bats

```diff
@@ -0,0 +1,220 @@
+    assert_failure
+    assert_output --partial $'received unrecognized \'--output FORMAT\' option of pickle; "plist" or "reg" must be specified'
+}
+
+@test 'report unrecognized registry sub-options' {
+    run rdctl create-profile --output=reg --hive=hklm --type=ruff --from-settings
+    assert_failure
+    assert_output --partial "invalid registry type of 'ruff' specified"
+}
+
+@test 'generates registry output for hklm/defaults' {
+    run rdctl create-profile --output reg --from-settings
+    assert_success
+    assert_output --partial '[HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Rancher Desktop\defaults\application]'
+
+    run rdctl create-profile --output reg --hive=hklm --from-settings
+    assert_success
+    assert_output --partial '[HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Rancher Desktop\defaults\application]'
+
+    run rdctl create-profile --output reg --hive=HKLM --type=Defaults --from-settings
+    assert_success
+    assert_output --partial '[HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Rancher Desktop\defaults\application]'
+
+    run rdctl create-profile --output reg --type=DEFAULTS --from-settings
+    assert_success
+    assert_output --partial '[HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Rancher Desktop\defaults\application]'
+}
+
+@test 'generates default registry output from inline json' {
+    run rdctl create-profile --output reg --body '{"application": { "window": { "quitOnClose": true }}}'
+    assert_success
+    assert_output --partial '[HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Rancher Desktop\defaults\application\window]'
+    assert_output --partial '"quitOnClose"=dword:1'
+}
+
+@test 'generates registry output for hklm/locked' {
+    run rdctl create-profile --output reg --hive=Hklm --type=Locked --from-settings
+    assert_success
+    assert_output --partial '[HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Rancher Desktop\locked\application]'
+    run rdctl create-profile --output reg --type=LOCKED --from-settings
+    assert_success
+    assert_output --partial '[HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Rancher Desktop\locked\application]'
+}
+
+@test 'generates registry output for hkcu/defaults' {
+    run rdctl create-profile --output reg --hive=Hkcu --from-settings
+    assert_success
+    assert_output --partial '[HKEY_CURRENT_USER\SOFTWARE\Policies\Rancher Desktop\defaults\application]'
+    run rdctl create-profile --output reg --hive=hkcu --type=Defaults --from-settings
+    assert_success
+    assert_output --partial '[HKEY_CURRENT_USER\SOFTWARE\Policies\Rancher Desktop\defaults\application]'
+}
+
+@test 'generates registry output for hkcu/locked' {
+    run rdctl create-profile --output reg --hive=HKCU --type=locked --from-settings
+    assert_success
+    assert_output --partial '[HKEY_CURRENT_USER\SOFTWARE\Policies\Rancher Desktop\locked\application]'
+}
+
+# The result of the `assert_output` for here-documents looks suspicious (I see it always passing),
+# but this serves to document the expected full reg output
```

Comment:
I don't see a here-document in this test, so maybe the comment is out-dated?

I would like to see an example of the test succeeding when it shouldn't, if you can still reproduce it! It looks a lot of tests would read better if they used a here-doc instead of a string of `assert_output --partial` lines.

Reply:
I pulled out the comments, and I'd rather do partial matching against `--from-settings`, but here-doc based matching against known JSON inputs, because then the outputs aren't subject to change after a change in default settings.

The test for registry output from `complex_json_data` doesn't need a `--partial` so there's a case of a full here-do c based test on registry output.

Reply:
I've added more here-docs. The secret was the `-` bit before the `<<TARGET`.